tests/handlers/test_sso.py

```diff
@@ -0,0 +1,127 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import BinaryIO, Callable, Dict, List, Optional, Tuple
+from unittest.mock import Mock
+
+from twisted.test.proto_helpers import MemoryReactor
+from twisted.web.http_headers import Headers
+from twisted.web.iweb import IResponse
+
+from synapse.http.client import RawHeaders, read_body_with_max_size
+from synapse.server import HomeServer
+from synapse.util import Clock
+
+from tests import unittest
+from tests.test_utils import SMALL_PNG, FakeResponse, simple_async_mock
+
+
+class TestSSOHandler(unittest.HomeserverTestCase):
+    def make_homeserver(self, reactor: MemoryReactor, clock: Clock) -> HomeServer:
+        self.http_client = Mock(spec=["request", "get_file"])
+        self.http_client.request.side_effect = mock_request
+        self.http_client.get_file.side_effect = mock_get_file
+        self.http_client.user_agent = b"Synapse Test"
+        hs = self.setup_test_homeserver(proxied_http_client=self.http_client)
+        return hs
+
+    async def test_set_avatar(self) -> None:
+        """Tests successfully setting the avatar of a newly created user"""
+        handler = self.hs.get_sso_handler()
+
+        # TODO: Create a new user to set avatar for
+        reg_handler = self.hs.get_registration_handler()
+        user_id = await reg_handler.register_user(approved=True)
+        # user_id = "@sso-user:test"
+
+        with self.assertLogs() as cm:
+            await handler.set_avatar(user_id, "http://my.server/me.png")
+        self.assertEqual(
+            cm.output, ["INFO:synapse.handlers.sso:successfully saved the user avatar"]
+        )
+
+        # TODO: ensure avatar returned via user's profile is SMALL_PNG
+        # profile_handler = self.hs.get_profile_handler()
+        # profile = await profile_handler.get_profile(user_id)
+        # profile["avatar_url"]
```

Comment:
This is now done. Since a new user is created, checking whether profile handler returns an avatar is enough (added this assert) and didn't need to compare bytes by bytes.